homeassistant/components/zwave_js/light.py

```diff
@@ -0,0 +1,329 @@
+        """
+        if self._target_value is not None and self._target_value.value is not None:
+            return round((self._target_value.value / 99) * 255)
+        if self.info.primary_value.value is not None:
+            return round((self.info.primary_value.value / 99) * 255)
+        return 0
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if device is on (brightness above 0)."""
+        return self.brightness > 0
+
+    @property
+    def hs_color(self) -> Optional[List[float]]:
+        """Return the hs color."""
+        return self._hs_color
+
+    @property
+    def white_value(self) -> Optional[int]:
+        """Return the white value of this light between 0..255."""
+        return self._white_value
+
+    @property
+    def color_temp(self) -> Optional[int]:
+        """Return the color temperature."""
+        return self._color_temp
+
+    @property
+    def min_mireds(self) -> int:
+        """Return the coldest color_temp that this light supports."""
+        return self._min_mireds
+
+    @property
+    def max_mireds(self) -> int:
+        """Return the warmest color_temp that this light supports."""
+        return self._max_mireds
+
+    @property
+    def supported_features(self) -> Optional[int]:
+        """Flag supported features."""
+        return self._supported_features
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the device on."""
+        # RGB/HS color
+        hs_color = kwargs.get(ATTR_HS_COLOR)
+        if hs_color is not None and self._supports_color:
+            # set white levels to 0 when setting rgb
+            await self._async_set_color("Warm White", 0)
+            await self._async_set_color("Cold White", 0)
+            red, green, blue = color_util.color_hs_to_RGB(*hs_color)
+            await self._async_set_color("Red", red)
+            await self._async_set_color("Green", green)
+            await self._async_set_color("Blue", blue)
+        else:
+            # turn off rgb when setting white values
+            await self._async_set_color("Red", 0)
+            await self._async_set_color("Green", 0)
+            await self._async_set_color("Blue", 0)
+
+        # Color temperature
+        color_temp = kwargs.get(ATTR_COLOR_TEMP)
+        warm: Optional[int] = None
+        if color_temp is not None and self._supports_color_temp:
+            # Limit color temp to min/max values
+            cold = max(
+                0,
+                min(
+                    255,
+                    round(
+                        (self._max_mireds - color_temp)
+                        / (self._max_mireds - self._min_mireds)
+                        * 255
+                    ),
+                ),
+            )
+            warm = 255 - cold
+            await self._async_set_color("Warm White", warm)
+            await self._async_set_color("Cold White", cold)
+            return
```

Comment:
Should we really return here? What if the service call contains more parameters, eg brightness?

Reply:
yeah, otherwise it will set the white light.
alternative is extra if statement for white_value

Reply:
We need to handle multiple parameters.

Reply:
that is not possible, is it ?
a light can't send both white value and color temp. At least, that's how I understand it.

Reply:
We need to handle all possible combinations. Brightness can be combined with other parameters, right?

Reply:
oh now I get what you mean. stupid mistake. Will fix it

Reply:
fixed